tt_metal/hw/inc/dev_msgs.h

```diff
@@ -234,13 +229,34 @@ struct watcher_msg_t {
     struct debug_ring_buf_msg_t debug_ring_buf;
 };
 
+struct dprint_buf_msg_t {
+    uint8_t data[DPRINT_BUFFERS_COUNT][DPRINT_BUFFER_SIZE];
+    uint32_t pad; // to 1024 bytes
+};
+
+// TODO: when device specific headers specify number of processors
+// (and hal abstracts them on host), get these from there (same as above for dprint)
+#if defined(COMPILE_FOR_ERISC) || defined (COMPILE_FOR_IDLE_ERISC)
+constexpr static std::uint32_t PROFILER_RISC_COUNT = 1;
+#else
+constexpr static std::uint32_t PROFILER_RISC_COUNT = 5;
+#endif
+
+struct profiler_msg_t {
+    uint32_t control_vector[kernel_profiler::PROFILER_L1_CONTROL_VECTOR_SIZE];
+    uint32_t buffer[PROFILER_RISC_COUNT][kernel_profiler::PROFILER_L1_VECTOR_SIZE];
+};
+
 struct mailboxes_t {
     struct ncrisc_halt_msg_t ncrisc_halt;
     struct slave_sync_msg_t slave_sync;
     uint32_t pad;
     struct launch_msg_t launch;
     struct watcher_msg_t watcher;
     struct dprint_buf_msg_t dprint_buf;
+    uint32_t pad1;
+    uint32_t pad2;
+    struct profiler_msg_t profiler;
```

Comment:
Is the padding here to ensure that the profiler struct base address respects NOC alignment?

Reply:
That is right.

Reply:
the downside of an array of pads is as we add/remove members the array size could be 0 which won't compile

Reply:
The size is chosen in the ifdef depending ERISC vs TENSIX. It is dependant on that because of the variable size of DPRINT and PROFILER.  I double checked and compile time size of zero is fine, it compiles.